### PR TITLE
1611 Make powershell exploiter interruptable

### PR DESCRIPTION
**Pull request by VakarisZ**
# What does this PR do?

Makes powershell exploiter interruptable

Add any further explanations here.

## PR Checklist
* [ ] Have you added an explanation of what your changes do and why you'd like to include them?
* [ ] Is the TravisCI build passing?
* [ ] Was the CHANGELOG.md updated to reflect the changes?
* [ ] Was the documentation framework updated to reflect the changes?

## Testing Checklist

* [x] Have you successfully tested your changes locally? Elaborate:
    > Tested by shutting down powershell machine, killing all monkeys
* [ ] If applicable, add screenshots or log transcripts of the feature working

## Explain Changes

Are the commit messages enough? If not, elaborate.

### CODE REVIEW

File: monkey/infection_monkey/exploit/powershell.py [hunk 1]
```diff
@@ -166,6 +176,9 @@ def _report_login_attempt(self, result: bool, credentials: Credentials):
     def _execute_monkey_agent_on_victim(self):
         monkey_path_on_victim = self.options["dropper_target_path_win_64"]
 
+        if self.is_interrupted():
+            raise self.InterruptError()
```
Comment (ilija-lazoroski) [inline monkey/infection_monkey/exploit/powershell.py:180]:
```suggestion
            raise self.InterruptError
```

File: monkey/infection_monkey/exploit/HostExploiter.py [hunk 2]
```diff
@@ -103,6 +103,10 @@ def is_interrupted(self):
             self.exploit_result.error_message = "Exploiter has been interrupted"
         return self.interrupt.is_set()
 
+    class InterruptError(Exception):
+        # Raise when exploiter gets interrupted
+        pass
+
```
Comment (ilija-lazoroski) [inline monkey/infection_monkey/exploit/HostExploiter.py:109]: This class should be under `common/utils/exceptions.py`, if we decide to go this route.

Comment (mssalvatore) [reply monkey/infection_monkey/exploit/HostExploiter.py:109]: It's not common. Nothing outside of exploiters uses it. Things should go in `common/` if, and only if, they're used (or expected to be used) by both agent and island

**Review by mssalvatore — COMMENTED** [review]
@VakarisZ @ilija-lazoroski Let's discuss this and #1789 on Monday. I'm not a fan of using exceptions for this. The interrupts are not an unexpected or unusual condition. Exploiters are aware that they can be interrupted and need to simply handle interruptions. Doing this is tantamount to a GOTO.

In general, we should consider a few things to improve the situation. Perhaps we should set an `interrupt` flag on `exploit_result` rather than setting the error message. This would achieve the same goal and use less data. We could also consider adding a callback instead of log message to `interruptable_iter`, which would make it more flexible.

We could also use a decorator to just skip the remaining methods once interrupt has been set. See #team-monkey for an example. 

Let's discuss on Monday over zoom.

**Review by mssalvatore — APPROVED** [review]

**Pull request merged**